src/engraving/dom/glissando.cpp

```diff
@@ -195,6 +205,41 @@ bool Glissando::pitchSteps(const Spanner* spanner, std::vector<int>& pitchOffset
     int direction = pitchEnd > pitchStart ? 1 : -1;
     pitchOffsets.clear();
     if (glissandoStyle == GlissandoStyle::DIATONIC) {
+        // Obey harp pedal diagrams if on a harp staff
+        if (glissando->isHarpGliss()) {
+            HarpPedalDiagram* hd = spanner->part()->currentHarpDiagram(spanner->tick());
+            std::set<int> playableTpcs = hd ? hd->getPlayableTpcs() : std::set<int>({ 14, 16, 18, 13, 15, 17, 19 });
+            std::vector<int> playablePitches;
+            for (int t : playableTpcs) {
+                playablePitches.push_back(tpc2pitch(t) % PITCH_DELTA_OCTAVE);
+            }
+
+            // Check for enharmonic on the next string
+            int en = noteStart->tpc() + TPC_DELTA_ENHARMONIC * -direction;
+            // Harp pedalling will only have 1 flat or sharp
+            if (en >= TPC_F_B && en <= TPC_B_S && playableTpcs.find(en) != playableTpcs.end()) {
+                pitchOffsets.push_back(0);
+            }
+            pitchOffsets.push_back(pitchStart - pitchStart);
```

Comment:
This line looks a bit weird (even though I do see the symmetry with line 229). If it's desired that `0` is pushed back, then I'd just write `0`, and perhaps add a comment why `0` might be pushed back twice (for example, say something like "Check for enharmonic on the next string"; this might cause `0` to be included twice")